src/components/views/elements/DialogButtons.tsx

```diff
@@ -69,17 +69,17 @@ export default class DialogButtons extends React.Component<IProps> {
     };
 
     private onCancelClick = (event: React.MouseEvent): void => {
-        this.props.onCancel(event);
+        this.props.onCancel!(event);
```

Comment:
In my opinion there should be a test that the `onCancel` is set [here](https://github.com/matrix-org/matrix-react-sdk/pull/10358/files#diff-46b24f2bf826b8f8dc45f660caef76a06fec748dd25a30b00c5d73cf6733c076R82).

Or this should better be 
```suggestion
        this.props.onCancel?(event);
```